### PR TITLE
browser: decrease constructor state

### LINKED ISSUE #12: Clean up `createDiffieHellman`
The constructor is a bit of a mess in terms of how it defaults certain arguments, maintaining a difficult-to-track cyclomatic complexity. 

It would be nice to reduce it down if possible.

### PR DESCRIPTION
**Pull request by dcousens**
Resolves #12.

Increases clarity IMHO,  especially since its just an argument shuffle.

### CODE REVIEW

File: browser.js [hunk 1]
```diff
@@ -10,11 +10,13 @@ function getDiffieHellman(mod) {
   return new DH(prime, gen);
 }
 
+var encodings = {
+  'binary': true, 'hex': true, 'base64': true
+}
```
Comment (dcousens) [inline browser.js]: Can revert back to `indexOf` if you think that is clearer.

File: browser.js [hunk 2]
```diff
@@ -10,11 +10,13 @@ function getDiffieHellman(mod) {
   return new DH(prime, gen);
 }
 
+var encodings = {
```
Comment (dcousens) [inline browser.js]: Maybe `ENCODINGS`?

File: browser.js [hunk 3]
```diff
@@ -10,11 +10,13 @@ function getDiffieHellman(mod) {
...
+  'binary': true, 'hex': true, 'base64': true
+}
+
 function createDiffieHellman(prime, enc, generator, genc) {
-  if (Buffer.isBuffer(enc) || (typeof enc === 'string' && ['hex', 'binary', 'base64'].indexOf(enc) === -1)) {
-    genc = generator;
-    generator = enc;
-    enc = undefined;
+  if (Buffer.isBuffer(enc) || encodings[enc] === undefined) {
+    return createDiffieHellman(prime, 'binary', enc, generator)
```
Comment (dcousens) [inline browser.js]: tl;dr: defaults to `binary` if the `enc` parameter is omitted.

**Comment by calvinmetcalf** [comment]
:+1:

File: browser.js (hunk 2 above)
Comment (calvinmetcalf) [inline browser.js]: yeah we're using it as a const

**Comment by calvinmetcalf** [comment]
looks good thanks for cleaning it up

**Comment by dcousens** [comment]
@calvinmetcalf up to you if you want to release after merge.

**Comment by calvinmetcalf** [comment]
published

**Pull request merged**